### ISSUE TITLE
Question about Cholesky parameterization in `decode_gaucho_offset`

### ISSUE THREAD

**Issue by FanChiMao** [jhlmarques/GauCho#4]
Hi, thanks for the implementation of Gaussian / Gaucho encoding.

I have a quick question regarding the Cholesky parameterization used in [`decode_gaucho_offset`](https://github.com/jhlmarques/GauCho/blob/main/mmrotate-gaucho/mmrotate/models/dense_heads/gaussian_fcos_head.py#L45-L63).

In the paper, the covariance matrix is defined via Cholesky decomposition as:

<img width="124" height="60" alt="Image" src="https://github.com/user-attachments/assets/2639797d-82f1-40ba-8fc4-cb3c12a64cb9" />

<img width="324" height="60" alt="Image" src="https://github.com/user-attachments/assets/4281fa1f-d078-4f94-858c-49246235c6ad" />
  

  
For reference, based on the paper, my understanding of the Cholesky-based covariance construction would be:

```python
alpha = cholesky[..., 0]
gamma = cholesky[..., 1]
beta  = cholesky[..., 2]

cov_a = alpha ** 2
cov_b = beta ** 2 + gamma ** 2
cov_c = alpha * gamma
```
  

However, in the current code:

```python
cov_a = cholesky[..., 0] ** 2
cov_b = (cholesky[..., 1] ** 2) + (cholesky[..., 2] ** 2)
cov_c = cholesky[..., 0] * cholesky[..., 2]
```
  
The off-diagonal term uses the third component instead of the second, which seems different from the paper formulation.

Is this an intentional modification for stability / training behavior, or could it be an index mismatch compared to the paper?

Thanks for any clarification!

**Comment by jhlmarques** [comment]
Hello, thanks for taking an interest in our project

I believe the misunderstanding is simply to how we index the elements; the relative order of the Cholesky elements we chose is $\alpha,\beta,\gamma$; mainly because $\alpha$ and $\beta$ are the diagonal, always positive elements. Some more reasoning behind this indexing is that alpha and beta are "more related" to the axis parameters $w,h$ and gamma is "more related" to orientation $\theta$, so we follow the common indexing of OBB parameters $w,h,\theta$.

This ordering of the elements can be seen in the forward pass in the same file:
```
gauss_xy = bbox_pred[:, 0:2] * stride
gaucho_ab = bbox_pred[:, 2:4].exp() * stride # cholesky a, b > 0
gaucho_c = self.conv_angle(reg_feat) * stride
(...)
bbox_pred = torch.cat([gauss_xy, gaucho_ab, gaucho_c], dim=1)
```

This ordering of the cholesky elements is also referenced in the paper, such as in the beginning of section 3.3.2
